### PR TITLE
Feature/26 service rewrite

### LINKED ISSUE #26: Support service rewrite mechanism
In the single-node CES, rewrites can be specified through the SERVICE_REWRITE environment variable in the Dockerfile:
```
SERVICE_REWRITE='{"pattern": "portainer", "rewrite": ""}'
```
This then gets generated into the following `rewrite` statement in the nginx configuration:
```
rewrite ^/portainer(/|$)(.*) /$2 break;
```
This behaviour should also be supported in multinode CES.

### PR DESCRIPTION
**Pull request by jelemux**
Resolves #26

### CODE REVIEW

File: controllers/ingressUpdater.go [hunk 1]
```diff
@@ -210,10 +236,17 @@ func (i *ingressUpdater) upsertDoguIsStartingIngressObject(ctx context.Context,
...
 	log.FromContext(ctx).Info(fmt.Sprintf("dogu is ready -> update ces service ingress object for service [%s]", service.GetName()))
+	serviceRewrite, err := cesService.generateRewriteConfig()
+	if err != nil {
+		return err
+	}
+
+	// This should overwrite the `Accept-Encoding: "gzip"` header that browsers send.
+	// Gzipping by dogus is a problem because it prevents the warp menu from being injected.
+	encodingOverwrite := "proxy_set_header Accept-Encoding \"identity\";"
+
```
Comment (nhinze23) [inline controllers/ingressUpdater.go:247]:
```
configurationSnippetFormat := "%s"
if serviceRewrite != "" {
    configurationSnippetFormat = "%s\n%s"
}
configurationSnippet := fmt.Sprintf(configurationSnippetFormat, encodingOverwrite, serviceRewrite)
```

File: controllers/ingressUpdater.go [hunk 2]
```diff
@@ -210,10 +236,17 @@ func (i *ingressUpdater) upsertDoguIsStartingIngressObject(ctx context.Context,
...
 	annotations := map[string]string{
-		// This should overwrite the `Accept-Encoding: "gzip"` header that browsers send.
-		// Gzipping by dogus is a problem because it prevents the warp menu from being injected.
-		ingressConfigurationSnippetAnnotation: "proxy_set_header Accept-Encoding \"identity\";",
+		ingressConfigurationSnippetAnnotation: fmt.Sprintf("%s\n%s", encodingOverwrite, serviceRewrite),
```
Comment (nhinze23) [inline controllers/ingressUpdater.go:249]:
```
		ingressConfigurationSnippetAnnotation:configurationSnippet,

File: controllers/ingressUpdater.go [hunk 3]
```diff
@@ -210,10 +236,21 @@ func (i *ingressUpdater) upsertDoguIsStartingIngressObject(ctx context.Context,
...
+	serviceRewrite, err := cesService.generateRewriteConfig()
+	if err != nil {
+		return err
+	}
+
+	// This should overwrite the `Accept-Encoding: "gzip"` header that browsers send.
+	// Gzipping by dogus is a problem because it prevents the warp menu from being injected.
+	encodingOverwrite := "proxy_set_header Accept-Encoding \"identity\";"
+
+	configurationSnippet := fmt.Sprintf("%s", encodingOverwrite)
```
Comment (cesmarvin) [inline controllers/ingressUpdater.go:248]: 🚫 **[checkstyle]** <gosimple> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>S1025: the argument is already a string, there's no need to use fmt.Sprintf

**Comment by nhinze23** [comment]
lgtm

**Pull request merged**